django/contrib/sessions/backends/signed_cookies.py

```diff
@@ -12,11 +15,26 @@ def load(self):
         raises BadSignature if signature fails.
         """
         try:
+            # Retrieve the session data if it contains valid data, no matter how old.
+            session_data_non_checked_age = signing.loads(
+                self.session_key,
+                serializer=self.serializer,
+                max_age=None,
+                salt='django.contrib.sessions.backends.signed_cookies',
+            )
+
+            max_age = session_data_non_checked_age.get('_session_expiry')
+            if isinstance(max_age, datetime.datetime):
+                max_age = max_age - timezone.now()
+
+            # Handle None and 0
+            if not max_age:
```

Comment:
Test coverage seems lacking. For example, commenting this if-block out doesn't result in any failures. Could you add some tests as needed for all the changes?